File: java/smartlist/src/main/java/com/smartlist/module/candidate/dto/response/CandidateRespDTO.java
package com.smartlist.module.candidate.dto.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class CandidateRespDTO {

    @Schema(description = "Candidate Id", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotNull
    @Positive
    Integer id;

    @Schema(description = "Candidate Name", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotBlank
    String name;

    @Schema(description = "Candidate Email", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotBlank
    @Email
    String email;

    @Schema(description = "Candidate Contact Number", example = "", requiredMode= Schema.RequiredMode.NOT_REQUIRED)
    @JsonProperty(value = "contact_number")
    String contactNumber;

    @Schema(description = "Is Candidate Verified", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotNull
    Boolean verified;

    @Schema(description = "Is Candidate Verified", example = "", requiredMode= Schema.RequiredMode.NOT_REQUIRED)
    @Positive
    @JsonProperty(value = "photo_id")
    Integer photoId;

}

File: javascript/smartlist/smartlist-api-mysql-project/src/modules/candidate/dto/response/candidate.resp.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { IsEmail, IsInt, IsNotEmpty, IsString } from 'class-validator';
import { Candidate } from 'src/entities/candidate.entity';

export class CandidateRespDto {
  @ApiProperty({
    type: Number,
  })
  @IsNotEmpty()
  @IsInt()
  id: number;

  @ApiProperty({
    type: String,
  })
  @IsNotEmpty()
  @IsString()
  name: string;

  @ApiProperty({
    type: String,
  })
  @IsNotEmpty()
  @IsEmail()
  email: string;

  @ApiProperty({
    type: String,
  })
  @IsNotEmpty()
  @IsString()
  contact_number: string;

  @ApiProperty({
    type: Boolean,
  })
  @IsNotEmpty()
  verified: boolean;

  @ApiProperty({
    type: Number,
  })
  @IsInt()
  photo_id: number;

  constructor(candidate: Candidate) {
    this.id = candidate.id;
    this.name = candidate.name;
    this.email = candidate.email;
    this.contact_number = candidate.contactNumber;
    this.verified = candidate.verified;
    this.photo_id = candidate.photo != null ? candidate.photo.id : null;
  }
}
